library/network_connections.py

```diff
@@ -1553,24 +1553,35 @@ def _complete_kwargs_loglines(self, rr, connections, idx):
                 c["persistent_state"],
             )
             prefix = prefix + (", '%s'" % (c["name"]))
-        for r in rr["log"]:
-            yield (r[2], "[%03d] %s %s: %s" % (r[2], LogLevel.fmt(r[0]), prefix, r[1]))
+        for severity, msg, idx in rr["log"]:
+            yield (
+                idx,
+                "[%03d] %s %s: %s" % (idx, LogLevel.fmt(severity), prefix, msg),
+                severity,
+            )
 
-    def _complete_kwargs(self, connections, kwargs, traceback_msg=None):
+    def _complete_kwargs(self, connections, kwargs, traceback_msg=None, fail=False):
         if "warnings" in kwargs:
-            logs = list(kwargs["warnings"])
+            warning_logs = list(kwargs["warnings"])
         else:
-            logs = []
+            warning_logs = []
 
+        debug_logs = []
         loglines = []
         for res in self._run_results:
             for idx, rr in enumerate(res):
                 loglines.extend(self._complete_kwargs_loglines(rr, connections, idx))
         loglines.sort(key=lambda x: x[0])
-        logs.extend([x[1] for x in loglines])
+        for x in loglines:
+            debug_logs.append(x[1])
+            if fail:
+                warning_logs.append(x[1])
+            elif x[2] not in (LogLevel.DEBUG, LogLevel.INFO):
```

Comment:
```suggestion
            elif x[2] < LogLevel.WARNING:
```
This allows to add other intermediate loglevels in the future since all log levels are numeric:
https://docs.python.org/3/library/logging.html#logging-levels

Reply:
It was >= instead of <, because we want to save as warning those logs with equal or higher priority than `WARN`. I've fixed it now!

Reply:
Good catch :-)